### PR TITLE
Add RegisterPlantCompDesignFlow for Refrigeration:Condenser:WaterCooled

### PR DESCRIPTION
**Pull request by JasonGlazer**
and \autosize to Energy+.idd. Addresses #5089

### CODE REVIEW

**Comment by JasonGlazer** [comment]
@EnergyArchmage and @mjwitte I made this change based on just emulating other places in the code that use RegisterPlantCompDesignFlow so I would appreciate if you look at the code change and let me know if I missed anything. I also add \autosize to the IDD since it seems necessary. I assumed that Condenser().DesVolFlowRate was the right variable to register but Condenser(). VolFlowRate is also set by this same input.

Here is the test object I used:

```
  Refrigeration:Condenser:WaterCooled,
    MediumTempCondenser,     !- Name
    58000.,                  !- Rated Effective Total Heat Rejection Rate {W}
    29.4,                    !- Rated Condensing Temperature {C}
    0.0,                     !- Rated Subcooling Temperature Difference {DeltaC}
    10.0,                    !- Rated Water Inlet Temperature {C}
    Condenser Inlet Node,    !- Water Inlet Node Name
    Condenser Outlet Node,   !- Water Outlet Node Name
    ConstantFlow,            !- Water-Cooled Loop Flow Type
    Cond Outlet Temp Sch,    !- Water Outlet Temperature Schedule Name
    autosize,                !- Water Design Flow Rate {m3/s}
    0.003,                   !- Water Maximum Flow Rate {m3/s}
    55.,                     !- Water Maximum Water Outlet Temperature {C}
    ,                        !- Water Minimum Water Inlet Temperature {C}
    ,                        !- End-Use Subcategory
    30.,                     !- Condenser Refrigerant Operating Charge Inventory {kg}
    65.,                     !- Condensate Receiver Refrigerant Inventory {kg}
    20.;                     !- Condensate Piping Refrigerant Inventory {kg}

```

Please let me know if this looks right.

**Comment by mjwitte** [comment]
@JasonGlazer For plant, the water flow rate should get registered regardless of whether it was autosized or entered as a hard value.  I don't think .DesVolFlowRate has a value yet here does it?  The flow needs to be registered later after any sizing has been done.

**Comment by JasonGlazer** [comment]
@mjwitte The Condenser( ).DesVolFlowRate only has a value based on user input at this point. I don't see that it ever really gets "sized" like other type of components. There is no "size" routine in RefrigeratedCase.cc and the GetInput is the only place that Condenser( ).DesVolFlowRate gets assigned in the file. I am not very familiar with this source file so please let me know if you think a code change needs to be in a different place. This is the location that @EnergyArchmage had suggested but maybe something else is needed to.

**Comment by mjwitte** [comment]
So, are you saying the field is _not_ autosizeable?

**Comment by JasonGlazer** [comment]
It was not autosizable. I assumed that the call to RegisterPlantCompDesignFlow was part of the autosizing process. I added \autosizable to the IDD as part of the checkin.

File: idd/Energy+.idd.in [hunk 1]
```diff
@@ -72963,6 +72963,7 @@ Refrigeration:Condenser:WaterCooled,
         \units m3/s
         \note note required units must be converted from L/s as specified in ARI 450-2007
         \minimum> 0.0
+        \autosizable
```
Comment (mjwitte) [inline idd/Energy+.idd.in:72966]: It's not autosizable, so this should be removed.

File: src/EnergyPlus/RefrigeratedCase.cc [hunk 2]
```diff
@@ -3339,16 +3339,21 @@ namespace RefrigeratedCase {
...
 							Condenser( CondNum ).VolFlowRate = Numbers( 5 );
-						} else {
+							tmpDesignWaterFlowRate = Condenser( CondNum ).DesVolFlowRate;
+						}
+						else {
 							ShowSevereError( RoutineName + CurrentModuleObject + "=\"" + Condenser( CondNum ).Name + "\" " + cNumericFieldNames( 5 ) + " must be greater than zero." );
 							ShowContinueError( "Revise flow rates." );
 							ErrorsFound = true;
 						}
+						RegisterPlantCompDesignFlow( Condenser( CondNum ).InletNode, tmpDesignWaterFlowRate );
```
Comment (mjwitte) [inline src/EnergyPlus/RefrigeratedCase.cc]: Why add tmpDesignWaterFlowRate here?  Could just use Condenser.DesVolFlowRate in the function call here.

**Comment by EnergyArchmage** [comment]
The call to RegisterPlantCompDesignFlow is part of autosizing but only at the plant loop level.  Need to roll up all the component sizes regardless of if the components themselves are autosizing or not.

File: src/EnergyPlus/RefrigeratedCase.cc (hunk 2 above)
Comment (JasonGlazer) [inline src/EnergyPlus/RefrigeratedCase.cc]: Yes, I was following the example used in other places in the code but they all had more logic involved with the tmp variables.

**Comment by JasonGlazer** [comment]
@EnergyArchmage @mjwitte how does this last commit look? Any further changes needed? If not, I will put together a test case.

**Comment by EnergyArchmage** [comment]
Yes, I have reviewed this change and it looks correct.  thanks.

**Comment by mjwitte** [comment]
@EnergyArchmage Once the CI reports come back, you can do the merge.

**Comment by mjwitte** [comment]
@EnergyArchmage  . . . and after a unit test gets added.

**Comment by EnergyArchmage** [comment]
Would you like me to work up a unit test?

**Comment by JasonGlazer** [comment]
If you have a specific one in mind that would be fine. I had just noticed that RefrigeratedCase.cc had very little code coverage so I was just going to test its GetInput. Let me know.

**Comment by EnergyArchmage** [comment]
No I haven't thought about much, just wasn't clear what mike's comment meant for me.  I think calling GetInput and then checking the contents of CompDesWaterFlow data structure would suffice.

**Comment by JasonGlazer** [comment]
@EnergyArchmage After talking with @mjwitte decided to go with a simpler unit test for RegisterPlantCompDesignFlow(). The GetInput on RefrigeratedCase test will have to wait for someone braver than me.

**Pull request merged**